packages/agent/src/agent/http/index.ts

```diff
@@ -1125,14 +1132,27 @@ export class HttpAgent implements Agent {
     return cbor.decode(await response.arrayBuffer());
   }
 
+  async getRootKey(): Promise<ArrayBuffer> {
+    return this.fetchRootKey();
+  }
+
   public async fetchRootKey(): Promise<ArrayBuffer> {
-    if (!this.#rootKeyFetched) {
-      const status = await this.status();
-      // Hex-encoded version of the replica root key
-      this.rootKey = (status as JsonObject & { root_key: ArrayBuffer }).root_key;
-      this.#rootKeyFetched = true;
+    if (!this.#shouldFetchRootKey) {
+      return this.#rootKey;
+    }
```

Comment:
If we should not fetch the root key, we return the existing root key, which is either a hard coded root key provided by the consumer or the mainnet root key.